src/main/java/jenkins/plugins/gerrit/GerritApiBuilder.java

```diff
@@ -117,26 +117,23 @@ public GerritApiBuilder stepContext(StepContext context)
 
   public GerritApi build() {
     GerritApi gerritApi = null;
-    List<HttpClientBuilderExtension> extensions = new ArrayList<>();
-    GerritAuthData authData = null;
     if (gerritApiUrl == null) {
       logger.println("Gerrit Review is disabled no API URL");
-    } else if (username == null) {
-      logger.println("Gerrit Review is disabled no credentials");
-      authData = new AnonymousAuth(gerritApiUrl.toString());
     } else {
-      authData = new GerritAuthData.Basic(gerritApiUrl.toString(), username, password);
+      List<HttpClientBuilderExtension> extensions = new ArrayList<>();
+      GerritAuthData authData = null;
+      if (username == null) {
+        authData = new AnonymousAuth(gerritApiUrl.toString());
+      } else {
+        authData = new GerritAuthData.Basic(gerritApiUrl.toString(), username, password);
+      }
       if (Boolean.TRUE.equals(insecureHttps)) {
         extensions.add(SSLNoVerifyCertificateManagerClientBuilderExtension.INSTANCE);
       }
+      gerritApi =
+          new GerritRestApiFactory()
+              .create(authData, extensions.toArray(new HttpClientBuilderExtension[0]));
     }
-    gerritApi =
-        new GerritRestApiFactory()
-            .create(authData, extensions.toArray(new HttpClientBuilderExtension[0]));
     return gerritApi;
   }
-
-  public boolean isAnonymous() {
```

Comment:
there is no benefit in this function as you could have just returned null from build if user is not provided

Reply:
Nope for two reasons:

1. Never use `null` to define behavior.  Code should tell what it does explicitly.
2. If you return a `null` API, how can you call Gerrit?